Add: HyRepl Init script loading

allison-casey opened the pull request:
Adds repl init scripts with `HYSTARTUP` env var. Closes #1832  
I think we should merge #1955 before this so that the `Configuration` heading in `repl.rst` is set properly.

Kodiologist commented:
You can probably construct an intersphinx link to the documentation for `PYTHONSTARTUP` instead of using a URL, although intersphinx always takes me some trial and error.

The added newline before `from hy.importer import runhy, HyLoader` is a typo, right?

I'd say it's counterintuitive to `pop` `repl_spy` and `repl_output_fn`. Leave them in the module.

allison-casey commented:
> The added newline before from hy.importer import runhy, HyLoader is a typo, right?
yes, fixed.

> I'd say it's counterintuitive to pop repl_spy and repl_output_fn. Leave them in the module.
fixed

> You can probably construct an intersphinx link to the documentation for PYTHONSTARTUP instead of using a URL
I could not for the life of me figure out how to get it to work. the section title for the env var is a bit wonky, maybe that's causing issues?

Kodiologist commented:
I'll take a stab.

Kodiologist commented:
Got it. It helps to grep through the result of `python3 -msphinx.ext.intersphinx https://docs.python.org/3/objects.inv`.

allison-casey commented:
oh dang i would never have guessed the additional `:envvar:` prefix. thanks!

scauligi commented on docs/language/repl.rst:
```diff
@@ -62,6 +62,59 @@ Once an exception has been thrown in an interactive session, the most recent exc
+
+  (setv
+    ;; Spy and output-fn will be set automatically for all hy repls
+    repl-spy False
```
minor nitpick, wouldn't `True` make more sense as an example?

allison-casey replied:
It would! I'll change it next time in front of my work computer

allison-casey replied:
done
